service/src/main/java/org/mskcc/smile/service/impl/SampleServiceImpl.java

```diff
@@ -84,6 +86,8 @@ public SmileSample fetchAndLoadSampleDetails(SmileSample sample) throws Exceptio
             }
             if (patientUpdated) {
```

Comment:
patient updated is just based on if there are any new aliases to add to the patient node and not if there has been an actual patient node swap

I would compare the existingPatient cmo patient id to the one from the incoming `sample.getPatient()`